### ISSUE TITLE
[2016.11.0rc2] looks temp files in /tmp for file operations

### ISSUE THREAD

**Issue by sjorge** [saltstack/salt#37786]
### Description of Issue/Question
Looking into a issue I noticed 2016.11.0rc2 (maybe earlier) is leaking files in /tmp

https://gist.github.com/sjorge/2aeed095393f06f6bde4c1a032b093c3
Too much data to be allowed in an issue!

```
[.]$ pfexec cat /tmp/tmpzzndCL
<?xml version='1.0'?>
<!DOCTYPE service_bundle SYSTEM '/usr/share/lib/xml/dtd/service_bundle.dtd.1'>
<service_bundle type='manifest' name='export'>
  <service name='pkgsrc/minihttpd' type='service' version='0'>
    <dependency name='network' grouping='require_all' restart_on='error' type='service'>
      <service_fmri value='svc:/milestone/network:default'/>
    </dependency>
    <dependency name='filesystem-local' grouping='require_all' restart_on='none' type='service'>
      <service_fmri value='svc:/system/filesystem/local:default'/>
    </dependency>
    <method_context>
      <method_credential group='root' user='root'/>
      <method_environment>
        <envvar name='LD_PRELOAD_32' value='/usr/lib/extendedFILE.so.1'/>
        <envvar name='PATH' value='/opt/local/sbin:/opt/local/bin:/sbin:/usr/sbin:/usr/bin'/>
      </method_environment>
    </method_context>
    <instance name='default' enabled='false'>
      <exec_method name='start' type='method' exec='/opt/local/sbin/minihttpd -p 8080 -d /salt/bootstrap/scripts' timeout_seconds='60'/>
      <exec_method name='stop' type='method' exec=':kill' timeout_seconds='60'/>
      <property_group name='startd' type='framework'>
        <propval name='duration' type='astring' value='child'/>
        <propval name='ignore_error' type='astring' value='core,signal'/>
      </property_group>
      <template>
        <common_name>
          <loctext xml:lang='C'>Python powered mini http server.</loctext>
        </common_name>
      </template>
    </instance>
  </service>
</service_bundle>
```

These files contain files that have been pushed to the minion in question through template. (jinja)
Each file is a different file, that was generated using a template...

```
[.]$  pfexec grep -r 'powered' /tmp
/tmp/tmpkFtQQi:          <loctext xml:lang='C'>Python powered mini http server.</loctext>
/tmp/tmpIcmKHW:          <loctext xml:lang='C'>Python powered mini http server.</loctext>
/tmp/tmpLbIIvD:          <loctext xml:lang='C'>Python powered mini http server.</loctext>
... truncated because way to much test ...
/tmp/tmpsP5iLB:          <loctext xml:lang='C'>Python powered mini http server.</loctext>
/tmp/tmpb5OkAa:          <loctext xml:lang='C'>Python powered mini http server.</loctext>
/tmp/tmp7qHEck:          <loctext xml:lang='C'>Python powered mini http server.</loctext>
```

It looks like a tmp file is left for every(?) state.apply run that has the template included in it!

The temp file remains, even if there are no content changes.

### Steps to Reproduce Issue
test.sls
```yaml
leaky-tmp:
  file.managed:
    - name: /tmp/leaky
    - source: salt://leaky.txt
    - template: jinja
    - context:
        name: world
```

leaky.txt
```
hello {name}
```

Run this state
```
[.]$ pfexec ls /tmp ; salt-call state.apply test ; salt-call state.apply test ; pfexec ls /tmp
leaky  ssh-XXXXfaasOP
local:
  Name: /tmp/leaky - Function: file.managed - Result: Clean Started: - 11:26:44.891201 Duration: 108.244 ms

Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time: 108.244 ms
local:
  Name: /tmp/leaky - Function: file.managed - Result: Clean Started: - 11:26:49.775186 Duration: 109.052 ms

Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time: 109.052 ms
leaky  ssh-XXXXfaasOP  tmp5axO3v  tmpnDMiHw
```

And we have 2 new fresh temp files 👎 

### Versions Report
```
Salt Version:
           Salt: 2016.11.0rc2

Dependency Versions:
           cffi: 1.4.2
       cherrypy: 3.8.0
       dateutil: 2.4.0
          gitdb: 0.6.4
      gitpython: 2.0.8
          ioflo: 1.6.5
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: 1.5.0
       M2Crypto: 0.22
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Sep 26 2016, 19:40:04)
   python-gnupg: 0.3.9
         PyYAML: 3.11
          PyZMQ: 14.4.1
           RAET: 0.6.5
          smmap: 0.9.0
        timelib: 0.2.4
        Tornado: 4.3
            ZMQ: 4.1.3

System Versions:
           dist:
        machine: i86pc
        release: 5.11
         system: SunOS
        version: Not Installed
```

**Comment by sjorge** [comment]
I did just clean my /tmp's as I was having issues due to this.

**Comment by sjorge** [comment]
@Ch3LL maybe you can have a look at this, I'm not sure if it is 2016.11.0rc2 specific or also effect develop/2016.3.

I updated the main comment with a simple reproducible state.

For now I have pushed a cron that cleans out /tmp/tmp* every week to keep it a bit under control.

**Comment by Ch3LL** [comment]
@sjorge I think you are running into this issue: #37541  which has been fixed here #37540, which has been added to the head of 2016.11. I went ahead adn made sure that fixed your issue using your test case, but if you could double check that it also fixes it for you as well that would be great.

**Comment by damon-atkins** [comment]
Hi @Ch3LL I think this needs to be in the 2016.11 release, even if 2016.11.0 is skipped and first release is 2016.11.1

**Comment by sjorge** [comment]
@Ch3LL I patched one of my minions, I will follow up tomorrow to see if it is fixed.

**Comment by Ch3LL** [comment]
@damon-atkins totally agree! The fix is already added and will be included with 2016.11.0 release just double checked. 

Also thanks @sjorge let us know :)

**Comment by sjorge** [comment]
@Ch3LL yep, that fixes the problem for me! I'm closing the issue.
